python/tchannel/scheme.py

```diff
@@ -41,10 +62,18 @@ def __init__(self):
     def type(self):
         return self._type
 
-    def deserialize(self, obj):
+    def deserialize_header(self, obj):
+        assert obj
```

Comment:
Also, nit: `obj` is a poor argument name for this case because we know it's a string. `s` or `string` would be preferable.